p2p/proto/consensus.proto

```diff
@@ -53,27 +53,29 @@ message ProposalInit {
 // The primary content of the block. This is opaque to the consensus module.
 message ProposalContent {
     oneof content {
-        Transactions  transactions        = 1;
-        BlockProof    proof               = 2;
-        // A list of precommits which supported the previous block. When `H+1` is decided this will
-        // become the canonical set of precommits for H which will determine the rewards.
-        //
-        // Note that the proof for `H` included in `H+K` or sent to L1 doesn't depend on this, as
-        // it merely commits us to the statement that "H had a quorum of precommits from Validators"
-        // without any commitment to the specific set of precommits used to prove that.
-        repeated Vote previous_precommits = 3;
+        // The timestamp field in metadata can impact consensus, specifically the lower bound
+        // applied. If nodes apply a lower bound validation based on their local time, then we
+        // risk a scenario where in round `R` proposal P is locked. Then in a later round the
+        // timestamp in P has gone stale. Therefore the lower bound should be "greater than the
+        // previous timestamp". Upper bounds don't suffer from this problem.
+        BlockMetadata metadata            = 1;
+        // Once block `H` is decided there remains a question; which set of validators receive a
+        // reward? More specifically, what is the canonical set of precommits for block `H`? Our
+        // current plan is for the proposer to set the first transaction in `H+1` to be writing the
+        // list of precommits for `H` to the staking contract in startknet.
+        Transactions  transactions        = 2;
+        BlockProof    proof               = 3;
     }
 
     // The sequence number of this content within the proposal.
     uint32 sequence_number                = 4;
 
-    // Proposal builder's signature on the above fields.
+    // Proposer's signature on the above fields.
     ConsensusSignature signature          = 5;
 }
 
 message ProposalFin {
```

Comment:
It may be confusing to others that Proposal is init, content, fin where content has more "stuff". maybe flatten?

Reply:
I made the change but I want to point some things out which give me pause:
- I'm not certain it's a problem for there to be more stuff within. Have we limited other messages due to it holding yet further messages? (I agree this is an enum to an enum so a bit different)
- It ties the message's structures. The thing I really notice is that for `ProposalFin` we want there to be the a proper Tendermint signature on the Proposal. Given the flattened structure though I had to move `sequence_number` and `signature` into the `Proposal` messaage. So I end up with another `signature` field inside of `ProposalFin`.